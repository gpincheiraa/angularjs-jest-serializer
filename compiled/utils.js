"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});
var removeHTMLComments = exports.removeHTMLComments = function removeHTMLComments(allChildrenNodes) {
    for (var i = 0; i < allChildrenNodes.length; i++) {
        if (allChildrenNodes[i].nodeType === 8) {
            allChildrenNodes[i].parentNode.removeChild(allChildrenNodes[i]);
        }
    }
    return allChildrenNodes;
};