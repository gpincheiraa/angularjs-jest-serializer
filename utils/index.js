export const removeHTMLComments = (allChildrenNodes) => {
    for(let i = 0; i < allChildrenNodes.length ; i++) {
        if(allChildrenNodes[i].nodeType === 8) {
            allChildrenNodes[i].parentNode.removeChild(allChildrenNodes[i]);
        }
    }
    return allChildrenNodes;
};
