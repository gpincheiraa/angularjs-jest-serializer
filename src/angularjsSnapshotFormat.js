import angular from "angular";
import { domAttributesList } from "./domAttributesList"
import { removeHTMLComments } from "./utils"

const angularInjector = angular.injector(["ng"]);
const $compile = angularInjector.get("$compile");
const $rootScope = angularInjector.get("$rootScope");

const HTMLNormalizer = (element) => {
    const allChildrenElements = element[0].getElementsByTagName("*");

    [...allChildrenElements].forEach((child) => {
		[...child.attributes].forEach((attr) => {
			if(attr.value && domAttributesList.indexOf(attr.name) === -1) {
                attr.value = "[ngExpression]";
            }
		});
	});
    return element[0];
}

export const angularjsSnapshotFormat = ({ template, $ctrl }) => {
    const scope = $rootScope.$new();
    scope.$ctrl = $ctrl;

    let el = $compile(template)(scope);
    scope.$digest();

    removeHTMLComments(el[0].childNodes);
    el = HTMLNormalizer(el);

    return el.outerHTML;
}
