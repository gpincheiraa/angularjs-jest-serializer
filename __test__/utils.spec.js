import { removeHTMLComments } from "../src/utils";

test("Should remove the comment elements in a given NodeList", () => {

    const bodyElement = document.querySelector("body");
    const divWithComments = document.createElement("div");
    divWithComments.innerHTML = `
        <!-- A annoying comment -->
        <div> <a> Remove the comments please" </a> </div>
    `;
    bodyElement.appendChild(divWithComments);

    removeHTMLComments(divWithComments.childNodes);

    expect(divWithComments.innerHTML).toMatchSnapshot();
})
