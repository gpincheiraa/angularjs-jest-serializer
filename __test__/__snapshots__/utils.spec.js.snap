// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should remove the comment elements in a given NodeList 1`] = `
"
        
        <div> <a> Remove the comments please\\" </a> </div>
    "
`;
